src/app/script/app_clipboard_object.cpp

```diff
@@ -0,0 +1,138 @@
+// Aseprite
+// Copyright (c) 2022-2024  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+
+#ifdef HAVE_CONFIG_H
+  #include "config.h"
+#endif
+
+#include "app/modules/palettes.h"
+#include "app/script/luacpp.h"
+#include "app/util/clipboard.h"
+#include "engine.h"
+#include "clip/clip.h"
+
+namespace app {
+namespace script {
+
+namespace {
+
+struct Clipboard {};
+
+int Clipboard_clear(lua_State* L)
+{
+  if (!clip::clear())
+    return luaL_error(L, "failed to clear the clipboard");
+
+  return 1;
+}
+
+int Clipboard_is_text(lua_State* L)
+{
+  lua_pushboolean(L, clip::has(clip::text_format()));
+  return 1;
+}
+
+int Clipboard_is_image(lua_State* L)
+{
+  lua_pushboolean(L, clip::has(clip::image_format()));
+  return 1;
+}
+
+int Clipboard_is_empty(lua_State* L)
+{
+  // Using clip::has(clip::empty_format()) had inconsistent results, might as well avoid false positives
+  // by just checking the two formats we support
+  lua_pushboolean(
+    L, !clip::has(clip::image_format()) && !clip::has(clip::text_format())
+  );
+  return 1;
+}
+
+int Clipboard_get_image(lua_State* L)
+{
+  doc::Image* image = nullptr;
+  doc::Mask* mask = nullptr;
+  doc::Palette* palette = nullptr;
+  doc::Tileset* tileset = nullptr;
+  const bool result = app::Clipboard::instance()->getNativeBitmap(
+    &image, &mask, &palette, &tileset
+  );
+
+  // TODO: If we get a tileset, should we convert it to an image?
```

Comment:
Good question, and probably this will make things a little harder. `set/getNativeBitmap()` will use a custom format to copy/paste images inside Aseprite/between Aseprite instances, and that custom format saves the original image format (e.g. indexed / tilemap) and the original palette/tileset.

This `app.clipboard.image` getter doesn't return this information. One option would be to return multiple values, in the same order, e.g.
```lua
local image, mask, palette, tileset = app.clipboard.image
```
But it's quite confusing.